packages/@aws-cdk/cdk/lib/duration.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * Represents a length of time.
+ */
+export class Duration {
+  /**
+   * @param amount the amount of Seconds the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Seconds.
+   */
+  public static seconds(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Seconds);
+  }
+
+  /**
+   * @param amount the amount of Minutes the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Minutes.
+   */
+  public static minutes(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Minutes);
+  }
+
+  /**
+   * @param amount the amount of Hours the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Hours.
+   */
+  public static hours(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Hours);
+  }
+
+  /**
+   * @param amount the amount of Days the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Days.
+   */
+  public static days(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Days);
+  }
+
+  /**
+   * @param duration an ISO-formtted duration to be parsed.
+   * @returns the parsed `Duration`.
+   */
+  public static parse(duration: string): Duration {
+    const matches = duration.match(/^PT(?:(\d+)D)?(?:(\d+)H)?(?:(\d+)M)?(?:(\d+)S)?$/);
+    if (!matches) {
+      throw new Error(`Not a valid ISO duration: ${duration}`);
+    }
+    const [, days, hours, minutes, seconds] = matches;
+    if (!days && !hours && !minutes && !seconds) {
+      throw new Error(`Not a valid ISO duration: ${duration}`);
+    }
+    return Duration.seconds(
+      _toInt(seconds)
+      + (_toInt(minutes) * 60)
+      + (_toInt(hours) * 3_600)
+      + (_toInt(days) * 86_400)
+    );
+
+    function _toInt(str: string): number {
+      if (!str) { return 0; }
+      return Number(str);
+    }
+  }
+
+  private readonly amount: number;
+  private readonly unit: TimeUnit;
+
+  private constructor(amount: number, unit: TimeUnit) {
+    if (amount < 0) {
+      throw new Error(`Duration amounts cannot be negative. Received: ${amount}`);
+    }
+
+    this.amount = amount;
+    this.unit = unit;
+  }
+
+  /**
+   * @returns the value of this `Duration` expressed in Seconds.
+   */
+  public toSeconds(_opts: TimeConversionOptions = {}): number {
+    switch (this.unit) {
+      case TimeUnit.Seconds:
+        return this.amount;
```

Comment:
Feels like this repeated lookup table should not be necessary. It's too easy to make mistakes in.

If we can express every unit in terms of seconds, we can calculate from every unit to every other one.

Reply:
That's a pretty good point!